### PR TITLE
Basic auth and v2 recatcha

### PR DESCRIPTION
**Pull request by tgorskievo**

### CODE REVIEW

File: functional-tests/www/index.js [hunk 1]
```diff
@@ -20,37 +21,53 @@ nunjucks.configure(app.get('views'), {
...
+app.get('/recaptcha', function (req, res) {
+  res.render('form/recaptcha.html')
+});
+
+app.post('/recaptcha',function(req,res){
+  if(req.body['g-recaptcha-response'] === undefined || req.body['g-recaptcha-response'] === '' || req.body['g-recaptcha-response'] === null) {
+    return res.json({"responseCode" : 1,"responseDesc" : "Please select captcha"});
+  }
+
+  res.render('index.njs')
```
Comment (tgorskievo) [inline functional-tests/www/index.js]: We do not validate that. All we test on this endpoint is if the `handleRecaptcha` step waits until the loader disappears.

File: functional-tests/www/index.js [hunk 2]
```diff
@@ -20,37 +21,53 @@ nunjucks.configure(app.get('views'), {
...
+})
+
+app.get('/recaptcha', function (req, res) {
+  res.render('form/recaptcha.html')
+});
+
+app.post('/recaptcha',function(req,res){
+  if(req.body['g-recaptcha-response'] === undefined || req.body['g-recaptcha-response'] === '' || req.body['g-recaptcha-response'] === null) {
+    return res.json({"responseCode" : 1,"responseDesc" : "Please select captcha"});
+  }
```
Comment (joszo) [inline functional-tests/www/index.js]: const array = [undefined,null, '']
if(array.includes(req.body['g-recaptcha-response'])){
    return res.json({"responseCode" : 1,"responseDesc" : "Please select captcha"});
}

Comment (sethii) [reply functional-tests/www/index.js]: Honestly, if you care about '' than if (array) is enough...

**Comment by simonsipl** [comment]
if I must hitch to something, what is v2 recatcha ;) ?

**Comment by tgorskievo** [comment]
> what is the point of this PR?

Well, I created this PR because Dawid already started implementing a step to handle basic auth. Konrad is working on handleRecaptcha v2, so I added it at the same time

File: functional-tests/www/index.js (hunk 2 above)
Comment (tgorskievo) [reply functional-tests/www/index.js]: well, in this case, it's enough to check just if `if (!req.body['g-recaptcha-response'])`

**Pull request merged**